add: Реворк кубика судьбы и магического слуги

L343-CJ opened the pull request:
<!-- Если вам необходимо отключить IconDiffBot2 или MapDiffBot2, тогда добавьте в название вашего пр-а тэги [IDB IGNORE] или [MDB IGNORE] соответственно. -->

<!-- Вы можете совмещать до 2-х тэгов, например, balance/tweak:
Список возможных тэгов для пр-а:
- add: Вы добавляете что-то новое.
- admin: Вы меняете что-то связанное с администрацией. (кнопки, управление, панели, щитспавн и т.д.)
- balance: Вы производите балансировку в игре.
- bugfix: Вы исправили некий баг.
- code_imp: Вы имплементируете новое для билда, не меняя при этом ничего в самой игре.
- config: Вы меняете конфиг или работу SQL.
- del: Вы удаляете что-то из игры.
- experiment: Ваш ПР сделан с целью какого-то эксперимента.
- map: Вы меняете только карту.
- image: Вы добавляете/удаляете спрайты.
- sound: Вы добавляете/удаляете звуки.
- spellcheck: Вы исправили опечатку.
- local: Вы добавляете новый текст на русском или переводите на русский.
- tweak: Вы внесли незначительную правку.
- refactor: Вы полностью переписали старый код, улучшив его, НО не изменив функционал.
- server: Вы меняете что-то связанное с серверной частью или Github.
- wip: Ваш PR в драфте и планируется длительная разработка. -->

## Что этот ПР делает
Изменяет работу кубика и магического слуги(одна из наград кубика). 

<details><summary>Новые награды с кубика(номер - выпавшее число)</summary>:

1. Damnatio memoriae. Мгновенная смерть персонажа, обращение куклы в пепел, вся память о нём уничтожается. Включает как записи экипажа и трудовой договор, так и цели антагонистов.
2. В торс наносится 100 брута острого урона со 100% бронепробитием. Органы вываливаются
3. Привет из Ада. Оглушает на 2 секунды. Призывает две разъяренные адские гончии и пять низших.
4. Опустошение. Убирает все надетые вещи, импланты (включая биоимпланты), восстанавливает гены к стандарту, убирает любую антажку.
5. Снижение скорости, бывшая шестая позиция.
6. Превращениие в низшую форму, бывшая пятая позиция.
7. Взрыв, бывшая восьмая позиция.
8. Оглушение на 5 секунд и нанесение 50 брута случайной конечности, с гарантированным переломом.
9. Случайный вирус из пула 7 биоугрозы.
10. Случайная медаль, кроме капитанской.
11. Бокс с разогретыми донк пакетами.
12. 5000 кредитов, бывшая тринадцатая позиция.
13. Капитанская запаска, бывшая восемнадцатая позиция.
14. Админ реджув, бывшая двенадцатая позиция.
15. Набор уники. Уника, два спидлоудера, кобура.
16. Две случайные книги заклинаний.
17. Уменьшение урона на 50%, бывшая девятнадцатая позиция.
18. Маяк синдиката (набор), бывшая семнадцатая позиция.
19. Волшебная монетка, про нее ниже.
20. Становление магом
</details>

Волшебная монетка:
При использовании запускает воут на магического слугу. Сам по себе он просто человек, привязанный к телу хозяина. НЕ получает урон хозяина и не переносит урон ему, лечится внутри его тела (в том числе на внутряки). При смерти возрождается через 5 минут. Может быть призван или отозван хозяином, но может и сам вылезать и залезать обратно. Все снаряжение остаётся на нём. При смерти хозяина - вылезает из его тела и оказывается смертен, если убьют до реанимациии хозяина - умрёт окончательно и бесповоротно, растворившись в лужу крови. И пока хозяина не возродят, не сможет залезть обратно в тело.
<!-- Опишите, что делает ваш пулл-реквест, укажите основные изменения. -->
<!-- В случае исправления бага, укажите ссылку на канал в #баг-репорты-v2. В ином случае, опишите баг и его воспроизведение. -->

## Почему это хорошо для игры
Улучшение наград кубика и забытой роли, которая никому нафиг не нужна была
<!-- Опишите, почему, по вашему, следует добавить эти изменения в билд. -->

## Демонстрация изменений
Да, я мог бы поставить тонну видосов изменений, но мне лень
<!-- Заполните, если Вы меняли спрайты, карту или ваши изменения требуют визуальной демонстрации изменений. -->
<details><summary>Демонстрации изменений</summary>
<p>

<!-- Разместите изображения внутри этого блока. -->

</p>
</details>

## Тестирование
Все проверено на локалке :catsmile:
<!-- Как Вы тестировали свои изменения? Ведь тестировали? -->

L343-CJ committed:
Реворк слуги

L343-CJ committed:
немного доработок

L343-CJ committed:
Merge branch 'master220' into kubiki

L343-CJ committed:
блябуду

L343-CJ committed:
Merge branch 'kubiki' of https://github.com/L343-CJ/Paradise into kubiki

L343-CJ committed:
линтер - макрос для genderize

L343-CJ committed:
пу

L343-CJ committed:
Локализация кубика судьбы

Не совсем финальный вариант т.к. буду менять грани, ну и соответственно их текст тоже

L343-CJ committed:
я долбаеб

THEROADCONEMAN commented:
А у меня вопрос, а к чему нужна эта позиция (уника и барабаны)  для условного неантага-не сбшника? Лучше заменить на общее но приятное, условный ген силы и регенерации без генетической тяжести.

Так же, стоит рассмотреть позицию с маяком синдиката, чисто пвпшная-манчерская тема (не  знаю уместно ли добавлять манчерская), мб тоже пересмотреть.
<img width="356" height="31" alt="image" src="https://github.com/user-attachments/assets/b4c9c59c-7b8e-4a16-9efb-b4d2e5fad1cf" />
<img width="428" height="35" alt="image" src="https://github.com/user-attachments/assets/dcead64d-c7f6-4d29-8466-38fcb2d8e976" />

L343-CJ committed:
Damnation

Сделана первая грань куба(единица), немного не работает замена цели + есть рантайм

L343-CJ committed:
Merge branch 'master220' into kubiki

L343-CJ committed:
Вроде починил, первая грань готова

L343-CJ committed:
Merge branch 'kubiki' of https://github.com/L343-CJ/Paradise into kubiki

L343-CJ committed:
вторая грань, butcher

L343-CJ committed:
minor mistake

L343-CJ committed:
Грань 3. Хз вроде работает но чет криво спавнит

L343-CJ committed:
теперь работает нормально

L343-CJ committed:
гончии больше не убивают друг друга

L343-CJ committed:
четвертая грань и микрофиксы

L343-CJ committed:
5 и 6 грани

L343-CJ committed:
7 грань, взрыв

L343-CJ committed:
грань восьмая, перелом

L343-CJ committed:
Грань 9. Вирус

L343-CJ committed:
кофе помогает от альцгеймера

L343-CJ committed:
11 грань, теплые донк покеты

L343-CJ committed:
12, 13 и 14 грани

L343-CJ committed:
все оставшиеся грани нах

L343-CJ committed:
Merge remote-tracking branch 'upstream/master220' into kubiki

L343-CJ committed:
fix linter

L343-CJ committed:
еще фикс блин

dageavtobusnick commented:
/gemini review

L343-CJ commented:
ебааать он насрал

L343-CJ commented on code/modules/mining/ores_coins.dm:
```diff
@@ -690,6 +690,63 @@ GLOBAL_LIST_INIT(sand_recipes, list(\
+	var/datum/mind/servant_mind = new /datum/mind()
+	servant_mind.transfer_to(Servant)
+	var/datum/antagonist/servant/serve = new(user)
+	servant_mind.add_antag_datum(serve)
```
Но ведь роль и цель выдаются ПОСЛЕ перезаписывания

L343-CJ replied:
Ты даже не отвечаешь бля? Железяка ебаная

L343-CJ replied:
Да и майнд же даже не перезаписывается, а выдается после того, как куклу уже присвоили игроку

Pelmenik123 commented:
Удали VSC анскилл

L343-CJ committed:
починил

PlayerUnknown14 commented:
Завтра гляну локализацию.

L343-CJ committed:
чертовы линтеры

PlayerUnknown14 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -235,49 +235,90 @@
 // Die of Fate
 /obj/item/dice/d20/fate
```
гендер не прописан

L343-CJ replied:
Так там у d20 прописан, тут просто такой же будет, не? Линтер же будет на одинаковое с родителем значение ругаться

PlayerUnknown14 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -235,49 +235,90 @@
 /obj/item/dice/d20/fate
 	name = "Die of Fate"
-	desc = "A die with twenty sides. You can feel unearthly energies radiating from it. Using this might be VERY risky."
+	desc = "Кубик с двадцатью гранями. От него исходит таинственная энергия. Использовать его может быть ОЧЕНЬ рисково."
```
```suggestion
	desc = "Кубик с двадцатью гранями. От него исходит таинственная энергия. Использовать его может быть ОЧЕНЬ рискованно."
```

PlayerUnknown14 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -235,49 +235,90 @@
 /obj/item/dice/d20/fate/cursed
 	name = "cursed Die of Fate"
-	desc = "A die with twenty sides. You feel that rolling this is a REALLY bad idea."
+	desc = "Кость с двадцатью гранями. Вы чувствуете, что бросать его ОЧЕНЬ плохая идея."
```
```suggestion
	desc = "Кость с двадцатью гранями. Вы чувствуете, что бросать её это ОЧЕНЬ плохая идея."
```

PlayerUnknown14 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -235,49 +235,90 @@
 	if(!used)
 		if(!ishuman(user) || !user.mind || (user.mind in SSticker.mode.wizards))
-			to_chat(user, span_warning("You feel the magic of the dice is restricted to ordinary humans!"))
+			to_chat(user, span_warning("Магией этого кубика могут воспользоваться только простые люди!"))
```
```suggestion
			to_chat(user, span_warning("[capitalize(declent_ru(NOMINATIVE))] отказывается вам подчиняться!"))
```

PlayerUnknown14 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -235,49 +235,90 @@
 
 		var/turf/T = get_turf(src)
-		T.visible_message(span_userdanger("[src] flares briefly."))
+		T.visible_message(span_userdanger("[capitalize(declent_ru(NOMINATIVE))] ярко вспыхива[PLUR_ET_YUT(src)]."))
```
```suggestion
		T.visible_message(span_userdanger("[capitalize(declent_ru(NOMINATIVE))] ярко вспыхива[PLUR_ET_YUT(src)]!"))
```

PlayerUnknown14 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -235,49 +235,90 @@
 
 	if(!ishuman(user) || !user.mind || (user.mind in SSticker.mode.wizards))
-		to_chat(user, span_warning("You feel the magic of the dice is restricted to ordinary humans! You should leave it alone."))
+		to_chat(user, span_warning("Магией этого кубика могут воспользоваться только простые люди! Вы должны оставить его здесь."))
```
```suggestion
		to_chat(user, span_warning("[capitalize(declent_ru(NOMINATIVE))] отказывается вам подчиняться!"))
```

PlayerUnknown14 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,256 @@
+			continue
+		for(var/datum/mind/user in obj.get_owners())
+			if(QDELETED(obj))
+				to_chat(user, span_userdanger("Вам кажется, что вы что-то забыли!"))
```
```suggestion
				to_chat(user, span_userdanger("Вам кажется, что вы что-то забыли..."))
```

PlayerUnknown14 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,256 @@
+		butchered.visible_message(
+			span_userdanger("Корпус [butchered] разваливается на части, и [GEND_HIS_HER(butchered)] компоненты вываливаются наружу!"),
+			span_userdanger("Вы не успели ничего осознать, как ваши части вывалились наружу."),
+			span_userdanger("Вы слышите звук вываливающихся запчастей и разрывающихся проводов.")
```
```suggestion
			span_userdanger("Корпус [butchered.declent_ru(GENITIVE)] разваливается на части, и [GEND_HIS_HER(butchered)] компоненты вываливаются наружу!"),
			span_userdanger("Ваши компоненты отваливаются от вашего корпуса прямо на глазах!"),
			span_userdanger("Вы слышите звук вываливающихся запчастей и разрывающихся проводов.")
```

PlayerUnknown14 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,256 @@
+	butchered.visible_message(
+		span_userdanger("Тонкая красная линия появляется на груди [butchered], и спустя мгновение [GEND_HIS_HER(butchered)] органы вываливаются наружу!"),
+		span_userdanger("Вы не успели ничего осознать, как ваши органы вывалились наружу."),
+		span_userdanger("Вы слышите звук вываливающихся органов.")
```
сейм

PlayerUnknown14 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,256 @@
+	)
 
+/obj/item/dice/d20/fate/proc/mob_swarm(mob/living/carbon/human/swarmed)
+	src.visible_message(span_userdanger("На месте кубика появился портал, из которого выходят адские отродья!"))
```
на месте [деклент]

PlayerUnknown14 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,256 @@
+	swarmed.Weaken(2 SECONDS)
+
+/obj/item/dice/d20/fate/proc/purify(mob/living/carbon/human/purified)
+	purified.visible_message(span_userdanger("[purified] выгляд[PLUR_IT_YAT(purified)] очистившим[PLUR_I(purified)]ся!"))
```
деклент

L343-CJ replied:
Так  разве у карбонов оно есть? Там имя куклы же ток

PlayerUnknown14 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,256 @@
+			purified.force_gene_block(gene.block, FALSE)
+
+/obj/item/dice/d20/fate/proc/slow(mob/living/carbon/human/slowed)
+	slowed.visible_message(span_userdanger("[slowed] начал[GEND_A_O_I(slowed)] двигаться медленнее!"))
```
[деклент] начинает ...

L343-CJ replied:
тот же вопрос

PlayerUnknown14 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,256 @@
+	slowed.add_movespeed_modifier(/datum/movespeed_modifier/die_of_fate)
+
+/obj/item/dice/d20/fate/proc/monkefy(mob/living/carbon/human/monkeyed)
+	monkeyed.visible_message(span_userdanger("[monkeyed] превраща[PLUR_ET_YUT(monkeyed)]ся в обезьяну!"))
```
деклент

PlayerUnknown14 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,256 @@
+	monkeyed.monkeyize()
+
+/obj/item/dice/d20/fate/proc/explode(mob/living/carbon/human/center)
+	center.visible_message(span_userdanger("Рядом с [center] происходит взрыв!"))
```
деклент

PlayerUnknown14 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,256 @@
+/obj/item/dice/d20/fate/proc/break_bone(mob/living/carbon/human/target)
+	var/obj/item/organ/external/limb = pick(target.bodyparts)
+	limb.fracture()
+	to_chat(target, span_userdanger("Вы чувствуете, как ваша [GLOB.body_zone[limb.limb_zone][NOMINATIVE]] трескается и ломается!"))
```
плюралайзы

PlayerUnknown14 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,256 @@
+	)
+	var/datum/disease/virus/D = new virus_type()
+	if(D.Contract(infected, is_carrier = TRUE))
+		to_chat(infected, span_danger("На секунду вам становится трудно дышать"))
```
```suggestion
		to_chat(infected, span_danger("На секунду вам становится трудно дышать!"))
```

PlayerUnknown14 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,256 @@
+	gifted.put_in_hands(id)
+
+/obj/item/dice/d20/fate/proc/revive(mob/living/carbon/human/revived)
+	revived.visible_message(span_boldnotice("[revived] выгляд[PLUR_IT_YAT(revived)] полностью здоров[GEND_YM_OI_YM_YMI(revived)]"))
```
деклент

PlayerUnknown14 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,256 @@
+	gifted.put_in_hands(box)
+
+/obj/item/dice/d20/fate/proc/books()
+	src.visible_message(span_userdanger("Две магические книги упали на пол!"))
```
падают

PlayerUnknown14 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,256 @@
+	new /obj/item/spellbook/oneuse/random(loc)
+
+/obj/item/dice/d20/fate/proc/resistance(mob/living/carbon/human/user)
+	user.visible_message(span_userdanger("[user] выгляд[PLUR_IT_YAT(user)] очень крепко!"))
```
деклент

PlayerUnknown14 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,256 @@
+	user.physiology.burn_mod *= 0.5
+
+/obj/item/dice/d20/fate/proc/random_bundle()
+	src.visible_message(span_userdanger("Появился подозрительный радио маяк!"))
```
появляется

PlayerUnknown14 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,256 @@
+	create_smoke(2)
+
+/obj/item/dice/d20/fate/proc/magic_coin(mob/living/carbon/human/gifted)
+	gifted.visible_message(span_userdanger("В руках у [gifted] появилась странная монета!"))
```
деклент, появляется

PlayerUnknown14 commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,256 @@
+	gifted.put_in_hands(coin)
+
+/obj/item/dice/d20/fate/proc/become_wizard(mob/living/carbon/human/wizard)
+	wizard.visible_message(span_userdanger("Потоки магической энергии вылетают из [declent_ru(GENITIVE)] в сторону [wizard]!"))
```
деклент

PlayerUnknown14 commented on code/modules/clothing/under/accessories/accessory.dm:
```diff
@@ -400,6 +400,9 @@
 
+/obj/item/clothing/accessory/medal/gold/nothing_award
+	name = "\"Nothing Ever Happens\" award"
+	desc = "Always bet on nothing."
```
А давай всё таки на русском

PlayerUnknown14 commented on code/modules/mining/ores_coins.dm:
```diff
@@ -690,6 +690,54 @@ GLOBAL_LIST_INIT(sand_recipes, list(\
+	. = ..()
+	if(!COOLDOWN_FINISHED(src, COIN_SUMMON_COOLDOWN))
+		return
+	to_chat(user, span_warning("Вы подкидываете монету в руке, и та начинает нагреваться"))
```
деклент

PlayerUnknown14 commented on code/modules/mining/ores_coins.dm:
```diff
@@ -690,6 +690,54 @@ GLOBAL_LIST_INIT(sand_recipes, list(\
+	COOLDOWN_START(src, COIN_SUMMON_COOLDOWN, 10 SECONDS)
+	var/list/mob/dead/observer/candidates = SSghost_spawns.poll_candidates("Вы хотите поиграть играть за слугу [user.real_name]?", ROLE_WIZARD, role_cleanname = "слугу", poll_time = 10 SECONDS, source = Servant)
+	if(!LAZYLEN(candidates))
+		to_chat(user, span_warning("Монета остывает у вас в руке. Возможно, стоит попробовать позже."))
```
деклент

L343-CJ committed:
Update code/game/objects/items/weapons/dice.dm

Co-authored-by: PlayerUnknown14 <139672964+PlayerUnknown14@users.noreply.github.com>

L343-CJ committed:
Update code/game/objects/items/weapons/dice.dm

Co-authored-by: PlayerUnknown14 <139672964+PlayerUnknown14@users.noreply.github.com>

L343-CJ committed:
Update code/game/objects/items/weapons/dice.dm

Co-authored-by: PlayerUnknown14 <139672964+PlayerUnknown14@users.noreply.github.com>

L343-CJ committed:
Update code/game/objects/items/weapons/dice.dm

Co-authored-by: PlayerUnknown14 <139672964+PlayerUnknown14@users.noreply.github.com>

L343-CJ committed:
Update code/game/objects/items/weapons/dice.dm

Co-authored-by: PlayerUnknown14 <139672964+PlayerUnknown14@users.noreply.github.com>

L343-CJ committed:
Update code/game/objects/items/weapons/dice.dm

Co-authored-by: PlayerUnknown14 <139672964+PlayerUnknown14@users.noreply.github.com>

L343-CJ committed:
Update code/game/objects/items/weapons/dice.dm

Co-authored-by: PlayerUnknown14 <139672964+PlayerUnknown14@users.noreply.github.com>

L343-CJ committed:
Update code/game/objects/items/weapons/dice.dm

Co-authored-by: PlayerUnknown14 <139672964+PlayerUnknown14@users.noreply.github.com>

L343-CJ committed:
насрал локализацией

L343-CJ committed:
Merge branch 'master220' into kubiki

L343-CJ committed:
скобки...

L343-CJ committed:
Merge branch 'kubiki' of https://github.com/L343-CJ/Paradise into kubiki

dageavtobusnick commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,256 @@
-			user.mind.make_Wizard()
+			become_wizard(user)
+
+/obj/item/dice/d20/fate/proc/Damnation(mob/living/carbon/human/damned)
```
Датумизируй

dageavtobusnick commented on code/game/objects/items/weapons/storage/boxes.dm:
```diff
@@ -306,6 +306,27 @@
+	)
+
+/obj/item/storage/box/warmdonkpockets/populate_contents()
+	for(var/I in 1 to 6)
```
С маленькой

L343-CJ committed:
фикс ревайва

L343-CJ committed:
фикс + правка для ИИ

L343-CJ committed:
еще правка и фикс

L343-CJ committed:
еще фиксы нейронки

L343-CJ committed:
последняя волна

PlayerUnknown14 commented on code/modules/clothing/under/accessories/accessory.dm:
```diff
@@ -400,6 +400,9 @@
 
+/obj/item/clothing/accessory/medal/gold/nothing_award
+	name = "награда \"Ничего Не Произошло\""
+	desc = "Всегда ставьте на то, что ничего не случиться."
```
```suggestion
	desc = "Всегда ставьте на то, что ничего не случится."
```

PlayerUnknown14 committed:
Update code/modules/clothing/under/accessories/accessory.dm

dageavtobusnick commented on code/game/gamemodes/objective.dm:
```diff
@@ -1736,3 +1736,15 @@ GLOBAL_LIST_EMPTY(admin_objective_list)
+/datum/objective/serve
+	name = "Служить"
+	antag_menu_name = "Служить"
+	var/mob/living/serve_to
```
переменные наследника после переменных родителя

dageavtobusnick commented on code/modules/paperwork/contract.dm:
```diff
@@ -24,6 +24,7 @@
 		return -1
 	target = nOwner.mind
 	update_text()
+	GLOB.employmentContracts += src
```
```suggestion
	GLOB.employmentContracts |= src
```

dageavtobusnick commented on code/modules/mining/ores_coins.dm:
```diff
@@ -690,6 +690,54 @@ GLOBAL_LIST_INIT(sand_recipes, list(\
+	if(!COOLDOWN_FINISHED(src, COIN_SUMMON_COOLDOWN))
+		return
+	to_chat(user, span_warning("Вы подкидываете [declent_ru(ACCUSATIVE)] в руке, и та начинает нагреваться"))
+	var/mob/living/carbon/human/Servant = new(user.loc)
```
с маленькой

dageavtobusnick commented on code/modules/antagonists/servant/servant_revive.dm:
```diff
@@ -0,0 +1,47 @@
+	Servant.equipOutfit(/datum/outfit/butler)
+	Servant.forceMove(master)
+
+	for(var/mob/ghost in GLOB.dead_mob_list)
```
тут разве могут быть не мобы в списке?

dageavtobusnick commented on code/modules/antagonists/servant/servant.dm:
```diff
@@ -0,0 +1,93 @@
+
+/datum/antagonist/servant/on_gain()
+	master_ref= WEAKREF(serve_to)
+	serv_ref = WEAKREF(owner.current)
```
викрефы нужны в экшенах. Тут то они тебе зачем?

dageavtobusnick commented on code/modules/antagonists/servant/servant.dm:
```diff
@@ -0,0 +1,93 @@
+	summon_action.servant = serv_ref.resolve()
+	self_summon_action.Grant(owner.current)
+	summon_action.Grant(serve_to)
+	var/datum/objective/serve/serve_obj= new(serve_to)
```
```suggestion
	var/datum/objective/serve/serve_obj = new(serve_to)
```

dageavtobusnick commented on code/modules/antagonists/servant/servant.dm:
```diff
@@ -0,0 +1,93 @@
+	if(owner.current.loc == serve_to)
+		in_owner = TRUE
+		ADD_TRAIT(owner.current, TRAIT_NO_BREATH, MAGIC_TRAIT)
+	else
```
через ретурн

dageavtobusnick commented on code/modules/antagonists/servant/servant.dm:
```diff
@@ -0,0 +1,93 @@
+/datum/antagonist/servant/proc/check_range()
+	SIGNAL_HANDLER
+
+	if(get_dist(serve_to, owner.current) <= 12)
```
что значит 12? магическое число

dageavtobusnick commented on code/modules/antagonists/servant/servant_revive.dm:
```diff
@@ -0,0 +1,47 @@
+
+/datum/servant_revive/proc/respawn()
+	var/servant_key
+	var/mob/living/carbon/human/Servant = new
```
нейминг

dageavtobusnick commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,238 @@
+	var/datum/data/record/damned_med_record = find_record("name", damned.real_name, GLOB.data_core.medical)
+	qdel(damned_sec_record)
+	qdel(damned_gen_record)
+	qdel(damned_med_record)
```
перепроверь, что записи корректно удаляются откуда нужно и не будет хардделов

L343-CJ replied:
<img width="484" height="192" alt="image" src="https://github.com/user-attachments/assets/37690c9e-1045-4a36-a13e-15faa4171b19" />
Ну вот в коде крио по сути тоже тупо qdel записей, только находят не проком а циклом

L343-CJ replied:
И да, записи удаляются, проверял

dageavtobusnick commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,238 @@
-			user.mind.make_Wizard()
+			become_wizard(user)
+
+/obj/item/dice/d20/fate/proc/Damnation(mob/living/carbon/human/damned)
```
разбей прок на несколько логических кусков + нейминг

dageavtobusnick commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,238 @@
+
+/obj/item/dice/d20/fate/proc/purify(mob/living/carbon/human/purified)
+	purified.visible_message(span_userdanger("[purified.declent_ru(NOMINATIVE)] выгляд[PLUR_IT_YAT(purified)] очистившим[PLUR_I(purified)]ся!"))
+	for(var/obj/item/I in purified)
```
нейминг

dageavtobusnick commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,238 @@
+/obj/item/dice/d20/fate/proc/purify(mob/living/carbon/human/purified)
+	purified.visible_message(span_userdanger("[purified.declent_ru(NOMINATIVE)] выгляд[PLUR_IT_YAT(purified)] очистившим[PLUR_I(purified)]ся!"))
+	for(var/obj/item/I in purified)
+		if(istype(I, /obj/item/organ))
```
из хелпер. Если нету - создай

dageavtobusnick commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,238 @@
+
+/obj/item/dice/d20/fate/proc/infect(mob/living/carbon/human/infected)
+	var/virus_type = pick(
+		1; /datum/disease/virus/anxiety,
```
Это что за говно? Зачем тут 1?

dageavtobusnick commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -286,137 +327,238 @@
+		1; /datum/disease/virus/wizarditis,
+		1; /datum/disease/virus/babylonian_fever
+	)
+	var/datum/disease/virus/D = new virus_type()
```
нейминг

L343-CJ committed:
Update code/modules/paperwork/contract.dm

Co-authored-by: dageavtobusnick <71216640+dageavtobusnick@users.noreply.github.com>

L343-CJ committed:
датумизировал

L343-CJ committed:
Merge branch 'kubiki' of https://github.com/L343-CJ/Paradise into kubiki

L343-CJ committed:
правки

L343-CJ committed:
правки

L343-CJ committed:
Update code/modules/antagonists/servant/servant.dm

Co-authored-by: dageavtobusnick <71216640+dageavtobusnick@users.noreply.github.com>

L343-CJ committed:
починил хуйню с викрефами

L343-CJ committed:
рантайм

L343-CJ committed:
правочки мои правки...

L343-CJ committed:
разбил damnation на проки

L343-CJ committed:
я ненавижу линтеры

L343-CJ committed:
пу-пу

L343-CJ committed:
последний рывок?

dageavtobusnick commented on code/datums/dice_of_fate_roll.dm:
```diff
@@ -0,0 +1,265 @@
+		if(19)
+			//Coin that can summon servant
+			magic_coin()
+		if(20)
```
эм. Ты не понял видимо смысл датумизации. Каждый эффект это отдельный наследник. Ты берешь датум родитель и переопределяешь его прок у каждого наследника, затем собираешь датумы в список по написанному тобой правилу с ключем в виде числа ролла и потом роллишь число и из списка получаешь датум по ключу. Список глобальный само собой

L343-CJ committed:
нормальная датумизация

dageavtobusnick commented on code/game/objects/items/weapons/dice.dm:
```diff
@@ -322,4 +322,8 @@
 /obj/item/dice/d20/fate/proc/effect(mob/living/carbon/human/user, roll)
-	new /datum/dice_roll(user, roll, src)
+	for(var/roll_val in GLOB.dice_rolls)
+		var/datum/dice_roll/rolled = new roll_val(user, src)
```
уже намного лучше. Но можно еще. Не нужно создавать на каждый ролл каждого кубика объект. Можно создать набор объектов, поместить их в кубик по индексам равным числу ролла. тоесть список из 20 элементов(new /list(максимальное число ролла)) и на каждом индексе будет уже созданный объект, который берется по индексу при ролле. user, src же передавать в activate()

L343-CJ committed:
Merge branch 'master220' into kubiki

L343-CJ committed:
залью для теста

L343-CJ committed:
я ебу

L343-CJ committed:
УРА ПОБЕДА

ROdenFL commented:
/gemini review

ROdenFL commented on code/modules/mining/ores_coins.dm:
```diff
@@ -690,6 +690,54 @@ GLOBAL_LIST_INIT(sand_recipes, list(\
+	servant.equipOutfit(/datum/outfit/butler)
+	servant.forceMove(user)
+	COOLDOWN_START(src, COIN_SUMMON_COOLDOWN, 10 SECONDS)
+	var/list/mob/dead/observer/candidates = SSghost_spawns.poll_candidates("Вы хотите поиграть играть за слугу [user.real_name]?", ROLE_WIZARD, role_cleanname = "слугу", poll_time = 10 SECONDS, source = servant)
```
Может не создавать человека, пока мы не уверены что кандидат есть?

L343-CJ replied:
Так это нужно чтобы иконка была, да и он потом все равно удаляется если нет кандидата

ROdenFL replied:
иконку можно пнгшкой сделать как будто просто, а тут время на создание хумана тратить, одного из самых жирных объектов в игре

L343-CJ replied:
Знал бы я как

ROdenFL commented on code/datums/dice_of_fate_roll.dm:
```diff
@@ -0,0 +1,282 @@
+	delete_records(user)
+	delete_from_objectives(user)
+	user.ghostize()
+	user.dust()
+	.=..()
+
+/datum/dice_roll/damnation/proc/delete_records(mob/living/carbon/human/user)
+	var/datum/data/record/user_sec_record = find_record("name", user.real_name, GLOB.data_core.security)
+	var/datum/data/record/user_gen_record = find_record("name", user.real_name, GLOB.data_core.general)
+	var/datum/data/record/user_med_record = find_record("name", user.real_name, GLOB.data_core.medical)
+	qdel(user_sec_record)
+	qdel(user_gen_record)
+	qdel(user_med_record)
+	for(var/obj/item/paper/contract/employment/contract as anything in GLOB.employmentContracts)
+		if(contract.target != user.mind)
+			continue
+		qdel(contract)
```
Плюс если удаляешь контракт то стоит его и из списка убрать

L343-CJ commented on code/modules/mining/ores_coins.dm:
```diff
@@ -690,6 +690,54 @@ GLOBAL_LIST_INIT(sand_recipes, list(\
+	. = ..()
+	if(!COOLDOWN_FINISHED(src, COIN_SUMMON_COOLDOWN))
+		return
+	to_chat(user, span_warning("Вы подкидываете [declent_ru(ACCUSATIVE)] в руке, и та начинает нагреваться"))
```
Эм что

L343-CJ committed:
теперь не qdel

L343-CJ committed:
финал

L343-CJ committed:
Я. Ненавижу. Линтер

ROdenFL reviewed (dismissed):
Мне не особо нравится кд на хил и я не особо разбираюсь в экшонах, в остальном впринципе ок

L343-CJ committed:
Merge branch 'master220' into kubiki

dageavtobusnick committed:
Update dice_of_fate_roll.dm

dageavtobusnick commented on code/datums/dice_of_fate_roll.dm:
```diff
@@ -0,0 +1,264 @@
+/datum/dice_roll/slow/activate(mob/living/carbon/human/user, obj/item/dice/d20/fate/dice)
+	user.visible_message(span_userdanger("[user.declent_ru(NOMINATIVE)] начал[GEND_A_O_I(user)] двигаться медленнее!"))
+	user.add_movespeed_modifier(/datum/movespeed_modifier/die_of_fate)
+	.=..()
```
```suggestion
	. = ..()
```

L343-CJ committed:
Update code/datums/dice_of_fate_roll.dm

Co-authored-by: dageavtobusnick <71216640+dageavtobusnick@users.noreply.github.com>

L343-CJ committed:
Merge branch 'master220' into kubiki

dageavtobusnick commented on code/modules/admin/admin_verbs.dm:
```diff
@@ -621,3 +621,27 @@ ADMIN_VERB(force_hijack, R_EVENT, "Toggle Shuttle Force Hijack", "Force shuttle
 	BLACKBOX_LOG_ADMIN_VERB("Shuttle Force Hijack")
+
+ADMIN_VERB_ONLY_CONTEXT_MENU(download_flaticon, R_ADMIN, "(Special) Download Icon", atom/thing in world)
+	var/icon/image = getFlatIcon(thing, no_anim = TRUE)
```
```suggestion
	var/icon/image = getFlatIcon(thing, no_anim = TRUE) //TODO replace with  iconforge flat icon
```

dageavtobusnick committed:
Apply suggestion from @dageavtobusnick
